File: kattis/magic_trick.pl
% https://open.kattis.com/problems/magictrick
% 1s
% 1.4 Easy

main :-
    read_line_to_codes(user_input,C),
    length(C,CL),
    sort(C,S),
    (length(S,CL)->T=1;T=0),
    writeln(T).

/*
% Compressed: 97 chars
main:-read_line_to_codes(user_input,C),length(C,CL),sort(C,S),(length(S,CL)->T=1;T=0),writeln(T).

*/

/*
main :-
    read_line_to_codes(user_input,C),
    length(C,CL),
    sort(C,S),
    length(S,SL),
    (CL =:= SL -> writeln(1) ; writeln(0)).
*/

/*
% Compressed: 103 chars
main:-read_line_to_codes(user_input,C),length(C,CL),sort(C,S),length(S,SL),(CL=:=SL->T=1;T=0),write(T).
*/

File: kattis/magic_trick.py
# https://open.kattis.com/problems/magictrick
# 1s
# 1.4 Easy

# 41 chars
print("01"[len(x:=input())==len(set(x))])


# 42 chars
# x=input()
# print("01"[len(x)==len(set(x))])


# 48 chars not short enough for Top 10 (16..39 chars)
# x=input()
# print(1 if len(x)==len(set(x)) else 0)
